backend/uclapi/dashboard/migrations/0008_populate_existing_secrets.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.1 on 2017-07-01 16:26
+from __future__ import unicode_literals
+
+from dashboard.app_helpers import generate_app_client_id, \
+    generate_app_client_secret
+
+from django.db import migrations
+
+
+def populate_client_data(apps, schema_editor):
+    App = apps.get_model('dashboard', 'App')
+    for app in App.objects.all():
+        app.client_id = generate_app_client_id()
+        app.client_secret = generate_app_client_secret()
+        app.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('dashboard', '0008_auto_20170701_1626'),

```

Comment:
is the fact that both this file and the previous file start with "0008" going to confuse Django? Or us for that matter?

Reply:
Nope! The file naming isn't actually relevant; it's done by the internal graph

Reply:
For the record, the reason I kept the 0008 here is that that migration is a multi-step process, but for our purposes, it should be treated as one step. Django just needs a helping hand to pre-populate programmatically generated default values in databases.

Reply:
ah ok! just fyi you can also do that in the same migration.

Example: https://github.com/uclapi/uclapi/blob/3a4bf4eac87a93b06d123ce7ffe7c314942ec2a7/backend/uclapi/dashboard/migrations/0008_auto_20170704_2240.py

Reply:
Ah, good to know. I wasn't sure if I could get a guarantee from Django that each of those operations would happen sequentially so I split it into another migration so that it would confirm that the table structure exists before it's populated. I'm fairly new to migrations hence the paranoia :P